diesel/src/backend.rs

```diff
@@ -396,6 +405,19 @@ pub(crate) mod sql_dialect {
 
         impl SupportsBatchInsert for PostgresLikeBatchInsertSupport {}
     }
+    /// This module contains all reusable options to configure
+    /// [`SqlDialect::ConcatClause`]
+    #[diesel_derives::__diesel_public_if(
+        feature = "i-implement-a-third-party-backend-and-opt-into-breaking-changes"
+    )]
+    pub(crate) mod concat_clause {
+
+        /// Indicates that this backend uses the
+        /// `CONCAT` syntax to select a concatenation
+        /// of two variables or string
+        #[derive(Debug, Clone, Copy)]
+        pub struct ConcatClause;
```

Comment:
I would prefer a name here that indicates that any backend using this impl uses `||` as concat operator. Additionally the documentation should also mention this explicitly.

Reply:
✅